packages/stylelint-plugin/src/util/variableUtils.ts

```diff
@@ -74,15 +81,17 @@ export const replaceTokenInVariable = (
  * Input: stylesheet,
  * Output: { "$foo": "red", "$other": "rgba(0, 0, 0, 0.1)". ...}
  */
-const getVariablesInBlock = (block: Container) =>
-  block.nodes
+const getVariablesInBlock = (block: ChildNode | Container | Document) => {
+  if (!("nodes" in block) || "toResult" in block) return {}
```

Comment:
@dmisdm 2 changes are made to add the `Document` type support into functions that process postcss variables as postcss parser and `parent` can return `Document` instead of `Container` or `ChildNodes`. 

I am not sure if my fix is valid as we might want to get `Doucment`'s global variables, which is causing a type error for a signature mismatch. As Document extends `Container<Root>`, it should able to be added without any issues.

The error stems from `.filter`

`
This expression is not callable.
  Each member of the union type '{ <S extends ChildNode>(predicate: (value: ChildNode, index: number, array: ChildNode[]) => value is S, thisArg?: any): S[]; (predicate: (value: ChildNode, index: number, array: ChildNode[]) => unknown, thisArg?: any): ChildNode[]; } | { ...; }' has signatures, but none of those signatures are compatible with each other.ts(2349)
`

```
return block.nodes
    .filter(
      (node): node is Declaration => node.type === "decl" && isVariable(node)
    )
    .reduce(
      (acc, next) => ({ ...acc, [next.prop]: next.value }),
      {} as Record<string, string | undefined>
    )
```

Reply:
@ckychris just an FYI that this `if (!("nodes" in block) || "toResult" in block) return {}` broke tests because it changed the behaviour at runtime, but i totally get why you did it

Reply:
at least i think it did...second guessing myself

Reply:
With the type restriction set to `Root` only. This line should be removed.

Reply:
unfortunately that breaks it! i think Document needs to be here, just working through it

Reply:
I think it should be ok now after the change i made, which checks for Document instances and bails out, for a good reason which is stipulated in the comments :)